src/components/objectListing/objectListing.component.tsx

```diff
@@ -0,0 +1,166 @@
+import {
+  createColumnHelper,
+  getCoreRowModel,
+  RowData,
+  useReactTable,
+} from "@tanstack/react-table";
+import clsx from "clsx";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
+import { Select } from "src/components/select";
+import { useListObjects } from "src/hooks/useListObjects";
+import { useSkylarkObjectTypes } from "src/hooks/useSkylarkObjectTypes";
+import { NormalizedObjectField } from "src/interfaces/skylark/objects";
+
+import { CreateButtons } from "./createButtons";
+import { RowActions } from "./rowActions";
+import { Table } from "./table";
+
+const ignoredKeys = ["__typename"];
+const orderedKeys = ["__typename", "title", "name", "uid", "external_id"];
+
+const columnHelper = createColumnHelper<object>();
+
+export type TableColumn = string;
+
+export interface ObjectListProps {
+  withCreateButtons?: boolean;
+}
+
+const createColumns = (columns: TableColumn[]) => {
+  return [
+    ...columns.map((column) =>
```

Comment:
again `[...[ ]]` isn't just the same as return map result?

Reply:
refactored to improve readability